.clj-kondo/hooks/metabase/api/common.clj

```diff
@@ -0,0 +1,45 @@
+(ns hooks.metabase.api.common
+  (:require
+   [clj-kondo.hooks-api :as api]
+   [clojure.string :as str]))
+
+(defn route-fn-name
+  "route fn hook"
+  [method route]
+  (let [route (if (vector? route) (first route) route)]
+    (-> (str (name method) route)
+        (str/replace #"/" "_")
+        symbol)))
+
+(defn defendpoint
+  [arg]
+  (letfn [(update-defendpoint [node]
+            (let [[_defendpoint method route & body] (:children node)]
+              (api/list-node
+               (list
+                (api/token-node 'do)
+                (api/token-node (symbol "compojure.core" (str (api/sexpr method))))
+                (-> (api/list-node
+                     (list*
+                      (api/token-node 'clojure.core/defn)
+                      (api/token-node (route-fn-name (api/sexpr method) (api/sexpr route)))
+                      body))
+                    (with-meta (meta node)))))))]
+    (update arg :node update-defendpoint)))
+
+(comment
+  (-> {:node (-> '(api/defendpoint POST "/:id/copy"
```

Comment:
Is there a blocker to using your recently added kondo testing mechanism instead?